libcaf_net/test/convert_ip_endpoint.cpp

```diff
@@ -0,0 +1,70 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#define CAF_SUITE convert_ip_endpoint
+
+#include "caf/detail/convert_ip_endpoint.hpp"
+
+#include "caf/test/dsl.hpp"
+
+#include <cstring>
+
+#include "caf/detail/socket_sys_includes.hpp"
+#include "caf/ipv4_endpoint.hpp"
+#include "caf/ipv6_endpoint.hpp"
+
+using namespace caf;
+using namespace caf::detail;
+
+CAF_TEST(sockaddr_in6 roundtrip) {
+  sockaddr_in6 source_addr = {};
+  source_addr.sin6_family = AF_INET6;
+  source_addr.sin6_port = htons(23);
+  source_addr.sin6_addr = in6addr_loopback;
+  auto ep = to_ip_endpoint(reinterpret_cast<sockaddr_storage&>(source_addr));
+  auto dest_addr = to_sockaddr(ep);
+  CAF_CHECK_EQUAL(memcmp(&source_addr, &dest_addr, sizeof(sockaddr_in6)), 0);
```

Comment:
Is bitwise equality really guaranteed without calling `memset` on the structs beforehand?

Reply:
I thought that initialising a struct like this: `sockaddr_in6 source_addr = {};` would initialize all members to their default values. With that in mind bitwise equality should be guaranteed without calling `memset`. Is that wrong?

Reply:
Seems like [you're right](https://en.cppreference.com/w/c/language/struct_initialization):

> All members that are not initialized explicitly are initialized implicitly the same way as objects that have static storage duration.

TIL. 👍